src/itemdef.cpp

```diff
@@ -230,6 +237,9 @@ void ItemDefinition::deSerialize(std::istream &is, u16 protocol_version)
 
 		sound_use.deSerialize(is, protocol_version);
 		sound_use_air.deSerialize(is, protocol_version);
+
+		has_on_place = readU8(is);
```

Comment:
`readU8` reads `0` without throwing an exception on EOF. Will this result in the expected bejaviour when connecting to old servers? If so, please add a comment above this line to clarify.

Reply:
This should only matter in case an outdated server is used with a new mod that adds the offhand list.

Which is a problematic case in general because if the offhand list is created by server-side Lua code, the client will assume the server supports dual wielding and render&use the offhand, even if the server C++ code doesn't support it.

Should the client be informed over network whether the server C++ code supports dual wielding or not, or should responsibility lie with the server admins to not add new mods that add the offhand list to older servers?

Note that server admins already carry the responsibility of ensuring all mods on the server are compatible with dual wielding before they add a mod that enables dual wielding.

Reply:
Proper functionality of the dual-wielding feature depends on server support, hence it makes sense to check against that by either `is.good()`, `is.eof()` or by increasing the protocol version.
Even though administrators are in charge for compatibility, there's no way to check against this feature from the mod API. Thus the idea to use the protocol version for that.

Reply:
anything more that was supposed to change here?